src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -2104,7 +2121,17 @@ skip:
         const uint32_t peer_stripe = tools::get_pruning_stripe(context.m_pruning_seed);
         const uint32_t local_stripe = tools::get_pruning_stripe(m_core.get_blockchain_pruning_seed());
         const size_t block_queue_size_threshold = m_block_download_max_size ? m_block_download_max_size : BLOCK_QUEUE_SIZE_THRESHOLD;
-        bool queue_proceed = nspans < BLOCK_QUEUE_NSPANS_THRESHOLD || size < block_queue_size_threshold;
+        m_span_limit = m_span_limit ? m_span_limit : BLOCK_QUEUE_NSPANS_THRESHOLD;
```

Comment:
> Why does this default to the large THRESHOLD value instead of the current MINIMUM value?

We want to utilize the hardware as much as we can.

> Also what's the thread-safety strategy for m_span_limit? A lock is held when it's set, but none is held during this read here. I think this will work if m_span_limit is an atomic, and cached here for the two reads.

Thanks for mentioning it. I missed it. I think changing it to atomic is the best approach.